### PR TITLE
[incubator-kie-drools#5743] Update Developer_Notes.md to be community…

### PR DESCRIPTION
**Pull request by tkobayas**
…-friendly

**Issue**: 

* https://github.com/apache/incubator-kie-drools/issues/5743

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

**Comment by tkobayas** [comment]
@gitgabrio @baldimir @yurloc @mariofusco Please review, thanks!

**Review by mariofusco — APPROVED** [review]

**Review by baldimir — APPROVED** [review]

File: drools-drl/drools-drl-parser/Developer_Notes.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# drools-drl-parser
+
+In this `dev-new-parser` branch, `drools-drl-parser` has 2 implementations of the DRL parser. The new parser is under development, and the old parser is used in the current drools code base.
+
+The old DRL parser is based on Antlr3 and contains lots of hard-coded logic in generated java codes, so it's hard to maintain. e.g. [DRL6Parser](https://github.com/apache/incubator-kie-drools/blob/main/drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/lang/DRL6Parser.java). It is the reason why the new parser is being developed. The old parser resources is placed under `src/main/java/org/drools/drl/parser/lang` and `src/main/resources/org/drools/drl/parser`.
```
Comment (yurloc) [inline drools-drl/drools-drl-parser/Developer_Notes.md]:
```suggestion
The old DRL parser is based on Antlr3 and contains lots of hard-coded logic in generated java codes, so it's hard to maintain e.g. [DRL6Parser](https://github.com/apache/incubator-kie-drools/blob/main/drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/lang/DRL6Parser.java). It is the reason why the new parser is being developed. The old parser resources are placed under `src/main/java/org/drools/drl/parser/lang` and `src/main/resources/org/drools/drl/parser`.
```

File: drools-drl/drools-drl-parser/Developer_Notes.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+## How to contribute to the development
+
+**Contribution would be highly appreciated!** Here is a rough guide to contribute to the development.
+
+1. Look at the parent issue [Experiment: New DRL Parser](https://github.com/apache/incubator-kie-drools/issues/5678) and choose a child issue which you want to work on.
+2. If you have an ASF committer role, assign yourself to the child issue. If you don't have the role, post a comment on the child issue that you are working on it so that we can avoid duplicated work.
+3. Create a feature branch based on the `dev-new-parser` branch.
+4. Fix the issue and make sure that the test in problem is passed and also all tests under `drools-drl-parser-tests` are passed.
+5. Add new tests to `drools-drl-parser-tests` to cover the issue. Hopefully, such a test would be a Descr comparison test. See `MiscDRLParserTest`.
```
Comment (yurloc) [inline drools-drl/drools-drl-parser/Developer_Notes.md]:
```suggestion
5. Add new tests to `drools-drl-parser-tests` to cover the issue. Hopefully, such a test would be a Descr comparison test. See `MiscDRLParserTest` and `DRLExprParserTest`.
```

**Comment by tkobayas** [comment]
Thank you for the suggestions, @yurloc !

**Comment by gitgabrio** [comment]
Hi @tkobayas , many thanks, that's great!
The only think I noticed is that this document dive deep in the current implementation (e.g old vs new approach) which is very good ( 👍  ) but reading it, I could not find a description of the "design" itself (beside that 

`DRLParser processes a DRL text and produces an AST(abstract syntax tree). Then apply DRLVisitorImpl to generate PackageDescr following the visitor pattern. So the main work would be implementing DRLParser.g4 and DRLVisitorImpl`

).
What I mean is that this document is very good to describe the current implementation, but does not provide a sort of mapping between the "technical needs"and the current code. Reason for raising this is that, in the future,
1. as we moved from different Antlr version, maybe we could switch to a completely different technology, and this document could help to distinguish what was the scope of different piece of code from what was the consequence
2. from general quality POV, it would help to verify if the current implementation, that is the result of long works and refactory, does exactly reflect the needs, or if it is flawed by some improper design choices that have been introduced long time ago and we are now keep building on top of.

Does this make sense ? Do you think it could be usefull/feasible to also provide this kind of information ?
Of course, it could be deferred to other ticket (and, of course, it is very possible that I misread/misundestood that document 😄  )

**Comment by tkobayas** [comment]
@gitgabrio Thank you for the suggestion. I updated with "Design Notes".

**Review by gitgabrio — APPROVED** [review]
Thanks @tkobayas LGTM!

**Pull request merged**